### PR TITLE
Kelsonic 19500 clp refinement

### PR DESCRIPTION
**Pull request by kelsonic**
## Description
**Issue:** https://github.com/department-of-veterans-affairs/va.gov-team/issues/19500

This PR does the following:

- [x] The short bright-blue border-bottom that accents a couple titles appears to have different lengths/widths. I think it has a relationship with the title above it.
- [x] Make sure the calendar event works when someone clicks the "add to calendar" link. I notice a runtime error about a missing description.
- [x] Change all layouts to consolidate all of the CMS-preview elements to the top of the page
- [x] Get the YouTube video working (does the content model include the "embed" link? we also need to adjust the width/height of the iframe because it doesn't do that automatically like an image)
- [x] Validate Hero Image Resolution is set appropriately. **FOLLOW HERE: https://dsva.slack.com/archives/C01DS1XDEQ0/p1613075910016200**
- [x] Log into ebenefits needs arrow icon at the end (Validate design specs - ask Ryan) **FOLLOW HERE: https://dsva.slack.com/archives/C01DS1XDEQ0/p1613072241009900**
- [x] Downloadable Resources - calls out for 'other languages' but doesn't have build-in on CMS to add or show on FE element.  _(Action Item - follow up with CMS)_ **FOLLOW HERE: https://dsva.slack.com/archives/C01DS1XDEQ0/p1613072382011100**
- [x] Events - Need to understand if something is virtual (zoom link) or other sources, how is that going to be displayed? (Should get the exact events from the event calendar.  FE needs to see exact data source example (Zoom link for example). **FOLLOW HERE: https://dsva.slack.com/archives/C01DS1XDEQ0/p1613075462013300**
- [x] Check margins on Mobile/Table some areas need adjusted for view.
- [x] Dynamic (blue, white) staggering background needs to be applied.  Do we want this into a separate ticket? 
- [x] Follow up, on https://github.com/department-of-veterans-affairs/va.gov-team/issues/16749

## Testing done
Wrote some unit tests, updated a nightwatch e2e test for coronavirus pages:

image

## Screenshots
localhost_3001_preview_nodeId=15068

localhost_3001_preview_nodeId=15068 (1)

## Acceptance criteria
- [x] View above

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/platform/site-wide/sass/modules/_m-campaign-landing-page.scss [hunk 1]
```diff
@@ -2,8 +2,12 @@
   width: 40px;
 }
 
+.va-c-blue-line--large {
+  width: 50px;
+}
+
 .va-u-box-shadow--none {
-  box-shadow: none;
+  box-shadow: none !important;
```
Comment (mpelzsherman) [inline src/platform/site-wide/sass/modules/_m-campaign-landing-page.scss:10]: I guess there's no `box-shadow` style in the design system?

File: src/site/filters/liquid.js [hunk 2]
```diff
@@ -593,4 +589,48 @@ module.exports = function registerFilters() {
...
+    }
+    if (hours) {
+      text = ' hours';
+    }
+
+    const digits = [hours, minutes, seconds].filter(digits => digits).join(':');
+
+    // Return a formatted timestamp string.
+    return `${digits}${text}`;
+  };
```
Comment (mpelzsherman) [inline src/site/filters/liquid.js:635]: Can we add unit tests for these? See `liquid.unit.spec.js`...

File: src/site/stages/build/drupal/graphql/nodeCampaignLandingPage.graphql.js [hunk 3]
```diff
@@ -507,6 +517,10 @@ module.exports = `
           fieldDescription
           fieldDuration
           fieldMediaVideoEmbedField
+          fieldPublicationDate {
+            date
+            value
+          }
```
Comment (mpelzsherman) [inline src/site/stages/build/drupal/graphql/nodeCampaignLandingPage.graphql.js:523]: Will you be making the transformer changes in another ticket?

File: src/platform/site-wide/sass/modules/_m-campaign-landing-page.scss (hunk 1 above)
Comment (kelsonic) [reply src/platform/site-wide/sass/modules/_m-campaign-landing-page.scss:10]: There is not ✅

File: src/site/stages/build/drupal/graphql/nodeCampaignLandingPage.graphql.js (hunk 3 above)
Comment (kelsonic) [reply src/site/stages/build/drupal/graphql/nodeCampaignLandingPage.graphql.js:523]: Maybe, we'll see

File: src/site/filters/liquid.js (hunk 2 above)
Comment (kelsonic) [reply src/site/filters/liquid.js:635]: Added! 🙂

Comment (mpelzsherman) [reply src/site/filters/liquid.js:634]: Great job on the test for `formatSeconds`. Do you think we need one for `createEmbedYouTubeVideoURL` as well?

Comment (kelsonic) [reply src/site/filters/liquid.js:634]: Done 👍

**Review by mpelzsherman — APPROVED** [review]

**Review by batemapf — APPROVED** [review]
👍 thanks for taking care of this!

**Review by zacharymorel — APPROVED** [review]

File: src/site/filters/liquid.js [hunk 4]
```diff
@@ -12,10 +12,6 @@ function getPath(obj) {
 module.exports = function registerFilters() {
   const { cmsFeatureFlags } = global;
 
-  // Set timeout option to something higher (20mins)
-  // eslint-disable-next-line no-new
-  new liquid.Context({ timeout: 1200000 });
-
```
Comment (ncksllvn) [inline src/site/filters/liquid.js:18]: Was this removal intentional? Seems like maybe it was lost during a merge resolution?

File: src/site/filters/liquid.js [hunk 5]
```diff
@@ -593,4 +589,47 @@ module.exports = function registerFilters() {
 
   liquid.filters.sortEntityMetatags = item =>
     item ? item.sort((a, b) => a.key.localeCompare(b.key)) : undefined;
+
+  liquid.filters.createEmbedYouTubeVideoURL = url => {
+    if (!url) {
+      return url;
+    }
```
Comment (ncksllvn) [inline src/site/filters/liquid.js:596]: If there isn't a URL, the parent component will hopefully never call this filter. I'm not opposed to even throwing an error in this case

File: src/site/filters/liquid.js [hunk 6]
```diff
@@ -593,4 +589,47 @@ module.exports = function registerFilters() {
...
+  };
+
+  liquid.filters.formatSeconds = rawSeconds => {
+    // Dates need milliseconds, so mulitply by 1000.
+    const date = new Date(rawSeconds * 1000);
+
+    // Derive digits.
+    const hours = date.getUTCHours() || '';
+    const minutes = date.getUTCMinutes() || '';
+    const seconds = date.getUTCSeconds() || '';
```
Comment (ncksllvn) [inline src/site/filters/liquid.js:616]: I wonder if we could get away with using Moment for this like `moment.unix(rawSeconds).format('hh:mm:ss')` -> `01:07:25`

File: src/site/includes/header.html [hunk 7]
```diff
@@ -86,6 +86,15 @@
...
 
 <body class="{{ body_class }} merger">
+  <!-- Draft status -->
+  {% if !entityPublished %}
+    <div class="vads-u-background-color--primary-alt-lightest vads-u-padding--1">
+      <div class="vads-l-grid-container medium-screen:vads-u-padding-x--0">
+        You are viewing a draft of "{{ entityUrl.path }}". <a data-same-tab href="{{ drupalSite }}/node/{{ entityId }}/edit">Edit this page in the CMS.</a>
+      </div>
+    </div>
+  {% endif %}
```
Comment (ncksllvn) [inline src/site/includes/header.html:96]: Love this. We maybe want to include `isPreview`.

```
{% if !entityPublished or isPreview %}

```

Or The `isPreview` just checks for the preview server and would make that banner show even if the page isn't a draft but is still being viewed as a preview

File: src/site/filters/liquid.js [hunk 8]
```diff
@@ -593,4 +589,47 @@ module.exports = function registerFilters() {
...
+
+    if (!_.includes(url, 'youtu')) {
+      return url;
+    }
+
+    if (_.includes(url, 'embed')) {
+      return url;
+    }
+
+    return _.replace(url, 'youtu.be', 'youtube.com/embed');
```
Comment (ncksllvn) [inline src/site/filters/liquid.js:606]: Nice and simple! I think I would have parsed the URL's domain with the node `url` module and way over-complicated it lol.

**Review by ncksllvn — APPROVED** [review]
Loving this!

File: src/site/filters/liquid.js (hunk 6 above)
Comment (zacharymorel) [reply src/site/filters/liquid.js:616]: Nic with the clutch moment api knowledge haha.

File: src/platform/site-wide/sass/modules/_m-campaign-landing-page.scss [hunk 9]
```diff
@@ -19,3 +23,11 @@
...
   text-transform: uppercase;
 }
+
+.va-c-video {
+  max-width: 560px;
+}
+
+.va-c-main > :nth-child(even) {
+  background: $color-primary-alt-lightest;
+}
```
[the remark this answers is not in the document]
Comment (kelsonic) [reply src/platform/site-wide/sass/modules/_m-campaign-landing-page.scss:33]: ya you would have to do ` > :nth-child` but definitely works!

File: src/site/filters/liquid.js (hunk 4 above)
Comment (kelsonic) [reply src/site/filters/liquid.js:18]: Ah true

File: src/platform/site-wide/sass/modules/_m-campaign-landing-page.scss (hunk 9 above)
[the remark this answers is not in the document]
Comment (ncksllvn) [reply src/platform/site-wide/sass/modules/_m-campaign-landing-page.scss:33]: ah up to you!

File: src/site/filters/liquid.js (hunk 5 above)
Comment (kelsonic) [reply src/site/filters/liquid.js:596]: I'm actually not going to throw an error, I'll just use the falsey value to conditional render the video element since they may not want to add one

File: src/site/filters/liquid.js (hunk 6 above)
Comment (kelsonic) [reply src/site/filters/liquid.js:616]: That's one way to do it! We'd have to clean the `00:00:0`s when there aren't seconds, so if it's ok I'll stick with my current solution 🙂

File: src/site/includes/header.html (hunk 7 above)
Comment (kelsonic) [reply src/site/includes/header.html:101]: Using `&&` for isPreview, seemed to work as desired locally and that way it doesn't show up on random pages 👍

File: src/platform/site-wide/sass/modules/_m-campaign-landing-page.scss (hunk 9 above)
[the remark this answers is not in the document]
Comment (kelsonic) [reply src/platform/site-wide/sass/modules/_m-campaign-landing-page.scss:33]: Updated! 🙂

File: src/site/filters/liquid.js (hunk 4 above)
Comment (kelsonic) [reply src/site/filters/liquid.js]: Updated! 🙂

**Comment by kelsonic** [comment]
Going to send up a follow-up PR with conditionally-rendered html sections: https://dsva.slack.com/archives/C01DS1XDEQ0/p1613157791037400

**Comment by kelsonic** [comment]
This icon bot is out of control... lol

File: src/site/includes/header.html (hunk 7 above)
Comment (ncksllvn) [reply src/site/includes/header.html:101]: Is `&&` correct? I thought Liquid used the word `and` lol

**Review by theodur — APPROVED** [review]

File: src/site/includes/header.html (hunk 7 above)
Comment (kelsonic) [reply src/site/includes/header.html:101]: I think I've seen both used so I wasn't sure which one to use, do you know by any chance?
